src/sentry/api/serializers/rest_framework/rule.py

```diff
@@ -184,13 +187,41 @@ def save(self, rule):
         return rule
 
 
+ACTION_UUID_KEY = "uuid"
+
+
+def ensure_action_uuid(action: dict[Any, Any]) -> None:
+    """
+    Ensure that each action is uniquely identifiable.
+    The function will check that a UUID key and value exists in the action.
+    If the key does not exist, or it's not a valid UUID, it will create a new one and assign it to the action.
+
+    Does not add an uuid to the action if it is empty.
+    """
+    if not action:
+        return
+
+    if ACTION_UUID_KEY in action:
+        existing_uuid = action[ACTION_UUID_KEY]
+        try:
+            UUID(existing_uuid)
+        except (ValueError, TypeError):
+            pass
+        else:
+            return
+
+    action[ACTION_UUID_KEY] = str(uuid4())
```

Comment:
This can be simplified:
```suggestion
    if action:
        try:
            UUID(action["uuid"])
            return
        except:
            action["uuid"] = str(uuid4())
        return
```

Reply:
Let me pull this into my other PR

Reply:
Super clean! Pulled the recommended changes to https://github.com/getsentry/sentry/pull/65276!